crates/stellar-scaffold-cli/src/commands/build/clients.rs

```diff
@@ -792,6 +810,87 @@ export default new Client.Client({{
         Ok(contract_id)
     }
 
+    async fn try_upgrade_contract(
+        &self,
+        name: &str,
+        existing_contract_id: Contract,
+        existing_hash: &str,
+        hash: &str,
+    ) -> Result<Option<Contract>, Error> {
+        let printer = self.printer();
+
+        let info_args = vec!["--wasm-hash", existing_hash, "--output", "json"];
+
+        let (_, existing_spec) = cli::contract::info::interface::Cmd::parse_arg_vec(&info_args)?
+            .get_spec(self.global_args.as_ref())
+            .await?;
+
+        let info_args = vec!["--wasm-hash", hash, "--output", "json"];
+
+        let (_, spec_to_upgrade) = cli::contract::info::interface::Cmd::parse_arg_vec(&info_args)?
+            .get_spec(self.global_args.as_ref())
+            .await?;
+
+        let legacy_upgradeable = Self::is_legacy_upgradeable(existing_spec);
+
+        if legacy_upgradeable.is_none() {
+            return Ok(None);
+        }
+
+        if Self::is_legacy_upgradeable(spec_to_upgrade).is_none() {
+            printer.warnln("New WASM is not upgradable. Contract will be redeployed instead of being upgraded.");
+            return Ok(None);
+        }
```

Comment:
```suggestion
        if !Self::is_legacy_upgradeable(existing_spec) {
            return Ok(None);
        }

        if !Self::is_legacy_upgradeable(spec_to_upgrade) {
            printer.warnln("New WASM is not upgradable. Contract will be redeployed instead of being upgraded.");
            return Ok(None);
        }
```

Reply:
`is_legacy_upgradeable` returns an option, it's `None` if contract is not upgradable, `Some(true)` if it's legacy upgradable (to be compatible with current version of the SEP), and `Some(false)` if it's newer upgradable contract (so it doesn't have an operator argument)
We can drop support of legacy in the future (but for now I think it's better to keep it so scaffold works on upgradable OZ contracts)